sdk/storage/azure-storage-common/src/xml_wrapper.cpp

```diff
@@ -392,19 +384,29 @@ namespace Azure { namespace Storage { namespace _internal {
 
 #else
 
-  struct XmlGlobalInitializer final
+  void XmlGlobalInitialize()
   {
-    XmlGlobalInitializer() { xmlInitParser(); }
-    ~XmlGlobalInitializer() { xmlCleanupParser(); }
-  };
+    static std::once_flag flag;
+    std::call_once(flag, [] { xmlInitParser(); });
+  }
 
-  static void XmlGlobalInitialize() { static XmlGlobalInitializer globalInitializer; }
+  void XmlGlobalDeinitialize()
+  {
+    static std::once_flag flag;
+    std::call_once(flag, [] { xmlCleanupParser(); });
```

Comment:
With the change in this PR, this cleanup function will never be called, right? Is this going to be a problem?

Reply:
It is a bit tricky ...

The code previously called
- `xmlInitParser` lazily when `XmlReader` or `XmlWriter` were initialized
- `xmlCleanupParser` when when singleton `XmlGlobalInitialize` is destroyed which happens at program end.

I would say this is generally fine.

Clickhouse (which uses azure-sdk-for-cpp) does quite aggressive testing with various Clang *sanitizer. One of them reported a memory leak in libxml at shutdown which was fixed [here](https://github.com/ClickHouse/ClickHouse/pull/45796). I also don't have more details except that there was some leak reported during our testing and that aforementioned PR fixes it.

Anyways, the fix requires a manual call of `XmlGlobalDeinitialize` which azure-sdk-for-cpp cannot expect of all using code in the general case.

To your point, I'll roll the libxml init/deinit code back to the previous state and we (ClickHouse) will continue to use azure-sdk-for-cpp with custom patching (yet the patch will be much smaller once this PR is merged).